src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -468,37 +473,42 @@ export class FileMatch extends Disposable implements IFileMatch {
 }
 
 export interface IChangeEvent {
-	elements: FileMatch[];
+	elements: (FileMatch | FolderMatch)[];
 	added?: boolean;
 	removed?: boolean;
 }
 
 export class FolderMatch extends Disposable {
 
-	private _onChange = this._register(new Emitter<IChangeEvent>());
+	protected _onChange = this._register(new Emitter<IChangeEvent>());
 	readonly onChange: Event<IChangeEvent> = this._onChange.event;
 
 	private _onDispose = this._register(new Emitter<void>());
 	readonly onDispose: Event<void> = this._onDispose.event;
 
-	private _fileMatches: ResourceMap<FileMatch>;
-	private _unDisposedFileMatches: ResourceMap<FileMatch>;
+	protected _fileMatches: ResourceMap<FileMatch>;
+	protected _folderMatches: ResourceMap<FolderMatchWithResource>;
+	protected _unDisposedFileMatches: ResourceMap<FileMatch>;
+	protected _unDisposedFolderMatches: ResourceMap<FolderMatchWithResource>;
 	private _replacingAll: boolean = false;
-
 	constructor(
 		protected _resource: URI | null,
 		private _id: string,
-		private _index: number,
-		private _query: ITextQuery,
-		private _parent: SearchResult,
+		protected _index: number | null,
```

Comment:
Should this just be on `FolderMatchWorkspaceRoot`?

Reply:
I think the reason I did this is because the sorting function (which takes a foldermatch) checks the index of folderMatches. So it's useful to also know if it's null.